lib/modules/manager/maven/extract.ts

```diff
@@ -76,6 +84,84 @@ function containsPlaceholder(str: string | null | undefined): boolean {
   return !!str && regEx(/\${[^}]*?}/).test(str);
 }
 
+function getCNBDependencies(
+  nodes: XmlElement[],
+  config: ExtractConfig,
+): PackageDependency[] {
+  const deps: PackageDependency[] = [];
+  nodes.forEach((node) => {
+    const depString = node.val.trim();
+    if (isDockerRef(depString)) {
+      const dep = getDockerDep(
+        depString.replace(DOCKER_PREFIX, ''),
+        true,
+        config.registryAliases,
+      );
+      dep.registryUrls = [];
+      dep.fileReplacePosition = node.position;
+      if (dep.currentValue || dep.currentDigest) {
+        deps.push(dep);
+      }
+    } else if (isBuildpackRegistryRef(depString)) {
+      const dep = getBuildpackDep(
+        depString.replace(BUILDPACK_REGISTRY_PREFIX, ''),
+      );
+
+      if (dep && (dep.currentValue || dep.currentDigest)) {
+        dep.registryUrls = [];
```

Comment:
same with registryUrls here

Reply:
if dep.registryUrls is not set, extract fails here: https://github.com/renovatebot/renovate/blob/c1d59fde1d831c6c8833c4d9774731fd7a9c55eb/lib/modules/manager/maven/extract.ts#L585

Reply:
Wouldn't that be wrong anyway or why would one want to add maven registries that were extracted from `settings.xml` to extrated docker image references? 

Perhaps it makes sense to make this safer and prevent adding additional registries by changing it to:
```ts
`if (dep.registryUrls) {
  dep.registryUrls.unshift(...additionalRegistryUrls);
}
```
...untested, so please verify

Reply:
Done, we had to touch this as well: https://github.com/renovatebot/renovate/blob/d02cc7028d9fc4bffd8e49db2dfd1a62a4942b1b/lib/modules/manager/maven/extract.ts#L553